airbyte-ci/connectors/connectors_insights/src/connectors_insights/hacks.py

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+from __future__ import annotations
+
+import json
+from typing import TYPE_CHECKING
+
+import requests
+from bs4 import BeautifulSoup
+from connectors_insights.consts import TEST_SUMMARY_ROOT_URL
+from google.cloud import storage
+
+if TYPE_CHECKING:
+    from typing import Dict, List
+
+    from connector_ops.utils import Connector  # type: ignore
+
+
+def get_ci_insights_from_json_report(json_report_url: str) -> List[str]:
+    storage_client = storage.Client(project="dataline-integration-testing")
+    bucket_name = json_report_url.split("/")[3]
+    blob_name = "/".join(json_report_url.split("/")[4:])
+    bucket = storage_client.bucket(bucket_name)
+    blob = bucket.blob(blob_name)
+
+    return json.loads(blob.download_as_string())
+
+
+def get_ci_on_master_report(connector: Connector) -> Dict | None:
```

Comment:
That would benefit from a docstring explaining why you're grabbing an html report here and what's in it.